mlflow/pyfunc/__init__.py

```diff
@@ -946,12 +959,18 @@ def load_model(
 
     _add_code_from_conf_to_system_path(local_path, conf, code_key=CODE)
     data_path = os.path.join(local_path, conf[DATA]) if (DATA in conf) else local_path
+
+    if isinstance(model_config, str):
+        model_config = _validate_and_get_model_config_from_file(model_config)
+
+    # conf.get(MODEL_CONFIG) this should ALWAYS be a dict (no more saving as file!)
     model_config = _get_overridden_pyfunc_model_config(
         conf.get(MODEL_CONFIG, None), model_config, _logger
     )
-
     try:
-        if model_config:
+        # currently, we do not support loading langchain with model_config like this
+        # langchain model_config must be specified using ModelConfig() class in code
```

Comment:
This is a TODO that we can remove right?

Reply:
Yes, I will change this to a TODO.